### DISCUSSION TITLE
Mac OS host appears to hang whilst compiling Flutter app for iOS

### DISCUSSION THREAD

**Discussion by dangilbert** [community/community#25025] [Actions]
<p>I’m using actions to build and deploy a flutter app and, whilst the android build works fine, the iOS build gets stuck.</p>
<p>The <code>.app</code> has been compiled using <code>flutter build ios --release --no-codesign</code> on the <code>macos-latest</code> environment.</p>
<p>I then attempt to sign it and compile the ipa for testflight using <code>fastlane build_ios_app</code> and this step hangs.</p>
<p>It appears that the compilation gets to the point of <code>Copying Flutter.framework</code> (which is a 500MB file, I assume because it includes the bitcode).</p>
<p>I’ve left the build running for 6 hours (the point where the build times out) and it never completes. The debug logging stops at the point of</p>
<pre><code>Copying /Users/runner/runners/2.164.0/work/x/x/ios/Flutter/Flutter.framework
##[debug]Re-evaluate condition on job cancellation for step: 'Fastlane Deploy iOS Testflight'.

##[error]The operation was canceled.

##[debug]System.OperationCanceledException: The operation was canceled.

##[debug] at System.Threading.CancellationToken.ThrowOperationCanceledException
</code></pre>
<p>Which doesn’t appear to contain anything useful. Has anyone managed to successfully build an iOS flutter project and deploy to testflight on github actions?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Please check if you have provided local code signing identities due to below error in your workflow. <a href="https://github.com/dangilbert/monica/runs/419693614?check_suite_focus=true#step:12:167" rel="nofollow noopener">https://github.com/dangilbert/monica/runs/419693614?check_suite_focus=true#step:12:167</a> The error occurs during action ‘maierj/fastlane-action’, it’s recommended to raise a query in the action issue here: <a href="https://github.com/maierj/fastlane-action/issues" rel="nofollow noopener">https://github.com/maierj/fastlane-action/issues</a>. Thanks.</p>

**Comment by dangilbert** [comment]
<p>Thanks! That appears to have been it. Having set up a separate keychain it now gets past that step and completes the signing.</p>
<p>It seems weird to me that it would keep going up to that step and then hang when the certificates weren’t installed correctly. It wasn’t even a step where signing should have been relevant.</p>
